community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/AbstractSchemaRule.java

```diff
@@ -22,21 +22,24 @@
 import java.nio.ByteBuffer;
 
 import org.neo4j.kernel.api.exceptions.schema.MalformedSchemaRuleException;
+import org.neo4j.kernel.api.schema.EntityPropertyDescriptor;
 import org.neo4j.storageengine.api.schema.SchemaRule;
 
 public abstract class AbstractSchemaRule implements SchemaRule, RecordSerializable
 {
     protected final Kind kind;
     protected final long id;
+    protected final EntityPropertyDescriptor descriptor;
 
-    public AbstractSchemaRule( long id, Kind kind )
+    public AbstractSchemaRule( long id, Kind kind, EntityPropertyDescriptor descriptor )
     {
         this.id = id;
         this.kind = kind;
+        this.descriptor = descriptor;
     }
 
     @Override
-    public long getId()
+    public final long getId()
```

Comment:
The id of a SchemaRule is a fixed property that should never be modified of overridden by subclasses, just like the getKind() method, which was already final.